### DISCUSSION TITLE
Changing graphname at runtime

### DISCUSSION THREAD

**Discussion by ghost** [JanusGraph/janusgraph#2471] [Q&A]
### Note: I cross-posted this to https://groups.google.com/g/gremlin-users so that I can reach a broader audience, so if you're a member of both groups, please receive my apologies!

Hello,
 
I have a situation where I have a consumer API that talks to JanusGraph. It's configured to connect to a a ConfiguredGraphFactory graph with graphname "Graph_A". I want to update Graph_A in *its totality*, which implies dropping it and recreating it from scratch. The problem is that such a process takes a long time, and I don't want the system to be down while Graph_A is being rebuilt. So I have another ConfiguredGraphFactory graph with graphname Graph_B. I take whatever time is required to create Graph_B, and when it's done, I need to stop the system, change the configuration of the API to now connect to Graph_B and restart the system.
 
But I don't want to do that.
 
Instead what I'd like to do is: when Graph_B is ready, I would ConfiguredGraphFactory.drop('Graph_A') and *rename* Graph_B to Graph_A.
 
Is that possible? If not, does anybody have another solution? This is akin to what one does in computer graphics and double buffering....

**Comment by dk-github** [comment]
Assuming that I understood correctly, and you are the designer/implementor of the API for consumers, on top of Janusgraph - why can't your API implementation switch which graph it is talking to, instead?..
you have a sequence of graphv1, graphv2, graphv3,...  and all you need to do when you create a new version is to tell your API to look at next version, then you can drop the previous one...
in this design, there is no need to rename anything in Janusgraph engine - you just change which graph name to look at...
